### DISCUSSION TITLE
Passing 'data' between Blazor pages

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#1649] [General]
Hi,
I have never liked passing parameters in url, what are your thoughts about this approach?
https://wellsb.com/csharp/aspnet/blazor-singleton-pass-data-between-pages/

Kind regards

**Comment by rockfordlhotka** [comment]
The thing you lose if you don't pass key values through the URL is the ability to do deep linking. The user won't be able to create a shortcut to a specific location in your app, and other apps or sites won't be able to link into your app.

**Comment by Chicagoan2016** [comment]
Thank you @rockfordlhotka, my concerns are around security (?), I was once working on a project and  looking at Sql Identity values in url didn't seem the best approach.

**Reply by rockfordlhotka** [reply]
Like passing a routing value such as `@page "personedit/{id : int}`?

I don't see the problem with this, as long as you use good hygene when building your query objects.

The problem with SQL injection, at least in .NET, occurs when you build statements as strings, but if you use a `SqlCommand` or equivalent type, they all have the ability to provide a collection of parameter values to the query - and the APIs always escape the parameter values so they aren't vulnerable to SQL injection.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka, Thank you, that was exactly my concern
